### ISSUE TITLE
Linter request: pytest "usefixture" mark should be flagged

### ISSUE THREAD

**Issue by lengau** [astral-sh/ruff#13795]
I've seen multiple cases where a test used `pytest.mark.usefixture` rather than `pytest.mark.usefixtures` in ways that caused the tests to only sometimes fail (e.g. only on certain platforms). It would be handy to have a linter that changed this.

E.g. the following code:

```python
@pytest.mark.usefixture("side_effect_fixture")
def test_my_function():
    assert my_function() == True
```

could be (probably with an `unsafe` marker?) converted to:

```python
@pytest.mark.usefixtures("side_effect_fixture")
def test_my_function():
    assert my_function() == True
```

**Comment by MichaReiser** [comment]
Do you have a reference to what `pytest.mark.usefixture`? I was only able to find the documentation for [`pytest.mark.usefixtures`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-mark-usefixtures) and it's not clear to me what the difference between the two is (I never used pytest)

**Comment by autinerd** [comment]
`@pytest.mark.usefixture` is almost always a typo and should emit a pytest warning. (but technically, a plugin could register the `usefixture` mark and do something with it)

Could you run `pytest <path_to_file> --markers` to check if `usefixture` is really a registered marker in your context?

**Comment by sbrugman** [comment]
Real-world examples:

- https://github.com/apache/superset/blob/master/tests/unit_tests/commands/dataset/test_update.py#L29
- https://github.com/bigchaindb/bigchaindb/blob/master/tests/tendermint/test_lib.py#L348

Reference issue at the pytest repository: https://github.com/pytest-dev/pytest/issues/3972

**Comment by lengau** [comment]
@MichaReiser as @autinerd said, it's almost always a typo. At runtime `pytest` will by default make an unknown mark warning. However, this popped up for me in a project where (unfortunately) the pytest runs are full of warnings, and it's faster/easier to fix stuff with ruff before tackling all of those warnings.
